### DISCUSSION TITLE
Build GPU code with GNU compilers

### DISCUSSION THREAD

**Discussion by sbrozell** [TinkerTools/tinker-gpu#161] [General]
What is the state of building the gpu code with gnu ?  See https://gcc.gnu.org/wiki/OpenACC

**Comment by jayponder** [comment]
We will certainly look into this at some point in the future. But keep in mind that the majority of Tinker9's key code (for speed) is in CUDA anyway. The OpenACC part is significantly less extensive and important, and has diminished over time as the code has been refined. We originally planned to make more use of OpenACC, but it just doesn't provide the fine-grained control needed to fully optimize the GPU code. And since we are making heavy use of CUDA anyway, it makes sense to focus on the NVIDIA compilers, which are freely available and (probably?) have the superior OpenACC implementation at present. And since you have to install and setup CUDA anyway, adding the NVIDIA compilers is really a very minor additional build requirement.

**Comment by sbrozell** [comment]
Thanks for the info.  I should have mentioned that my perspective is from the installation standpoint.   While you have good documentation for ensuring that the PGI compilers have appropriate GNU tools under the hood, for those based on redhat with old gnu compilers several prerequisite steps are needed before building tinker9.  For most users this is probably a non issue, but at an HPC center, it is nontrivial to get old redhat based pgi up to snuff (admittedly such pgi compilers could have been considered broken).  This combined with the now not so recent history of PGI led me to inquire about an all GNU build route for the OpenACC code.

**Comment by zhi-wang** [comment]
I'll move this issue to the discussion section.
